### ISSUE TITLE
When configuring dd-trace to use a logger, the passed in logger's level is manipulated.

### ISSUE THREAD

**Issue by blaines** [DataDog/dd-trace-rb#681]
In the case of configuring the datadog trace client, when passing a custom logger into the datadog client the custom logger's level will be changed to WARN. This had a negative effect on our system, hiding all INFO logs. Example of our configuration:
```
    Datadog.configure do |c|
      c.tracer enabled: true, env: env_name, hostname: host, port: port, log: logger, debug: debug
      # etc...
```
Once Datadog trace is configured `debug_logging=` will be called here:
https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/configuration.rb#L49

Because `debug = false` (a default) the log level for the passed in custom logger will be changed to WARN here:
https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/tracer.rb#L57

My suggestion would be to keep the same behavior only if the detected logger configured is a Datadog::Logger otherwise noop. Thoughts?

**Comment by Mange** [comment]
This hit us too. I had to monkey patch `Rails.logger.level=` to print a stack trace in production in order to find the cause of it as I couldn't reproduce it anywhere else.

**Comment by delner** [comment]
This should be fixed with #721, and will roll in 0.21.0. Please let me know if this works for you!
